### PR TITLE
Issue #10  As a user, I want to be alerted when I’m entering an empty item or an item that's already on my list

### LINKED ISSUE #10: 10. As a user, I want to be alerted when I’m entering an empty item or an item that's already on my list
## Summary
Users shouldn't be able to add an empty item to their list, or add the same item twice. If they try to do this, we need to show them an error message that explains the problem. This way, we’ll prevent some clutter in their lists.

## Acceptance criteria
- [x] Show an error message if the user tries to submit an empty item
- [x] Show an error message if the user tries to submit a new item that is _identical_ to an existing name. For instance, if the list contains `apples` and the user adds `apples`.
- [x] Show an error message if the user tries to submit a new item that matches an existing name with punctuation and casing normalized. For instance, if the list contains `apples` and the user adds `aPples` or `apples,` or `a pples`.
- [x] The user’s original input is saved in the database

##  Notes
- There are a couple of ways to approach this task. If you’re not sure how to begin, _ask your mentors_ for advice!

### PR DESCRIPTION
**Pull request by jmahamed**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

The code adds error handling to the manage list page. If the correct input is not added then a relevant error message will appear

## Related Issue

closes issue 10

## Acceptance Criteria

-  Show an error message if the user tries to submit an empty item
-  Show an error message if the user tries to submit a new item that is identical to an existing name. For instance, if the list contains apples and the user adds apples.
-  Show an error message if the user tries to submit a new item that matches an existing name with punctuation and casing normalized. For instance, if the list contains apples and the user adds aPples or apples, or a pples.
-  The user’s original input is saved in the database

## Type of Changes

error handling

### Before

Screenshot 2024-03-09 083151

### After

Screenshot 2024-03-09 083052

## Testing Steps / QA Criteria

try entering an empty item, the same item that you already have on your list and the same item with punctuation or capitalization or an empty space

### CODE REVIEW

**Review by IamHenryOkeke — COMMENTED** [review]
Feature works great.

File: src/views/ManageList.jsx [hunk 1]
```diff
@@ -18,17 +18,50 @@ export function ManageList({ listPath, userId }) {
 		theme: 'light',
 	};
 
+	// Fetch existing items from the list
+	const existingItems = useShoppingListData(listPath);
+
```
Comment (redapy) [inline src/views/ManageList.jsx]: Nice use of the custom hook to fetch the data!  We already have the data fetched and ready in the `App.jsx` file, you can just pass it as a prop to the `ManageList` components.

File: src/views/ManageList.jsx [hunk 2]
```diff
@@ -18,17 +18,50 @@ export function ManageList({ listPath, userId }) {
...
+			return;
+		}
+
+		// Add the new item to the database
+
 		try {
 			await addItem(listPath, {
-				itemName,
+				itemName: normalizedItemName, // Pass normalized item name
+				originalItemName: itemName, // Pass original item name
```
Comment (redapy) [inline src/views/ManageList.jsx]: the field `originalName` is unnecessary. If the item does not exist, we should save the name as entered by the user

**Review by redapy — CHANGES_REQUESTED** [review]
Great work! There are only some minor changes that should be implemented. Also, there is a missing case: if there is for example, an existing item `apples` and the user enters `a pples` it will be added but it should not

File: src/views/ManageList.jsx (hunk 2 above)
Comment (ijayhub) [reply src/views/ManageList.jsx]: The acceptance criteria state that we need to store the `originalName` first. That's why I made sure to define it before proceeding with normalization.

File: src/views/ManageList.jsx (hunk 1 above)
Comment (ijayhub) [reply src/views/ManageList.jsx]: so the useShoppingListData is unnecessary

Comment (ijayhub) [reply src/views/ManageList.jsx]: resolved... have seen it... thanks

**Pull request merged**